### PR TITLE
feat: add utilities for loadtests

### PR DESCRIPTION
**Pull request by Zygimantass**
# In this PR

Adds utilities for creating a `tm-load-test` loadtest using Petri abstractions (InteractingWallet, ChainClient, etc.)

# How to use this

I have an example of how this can be used in the BlockSDK suite:

```go
	var endpoints []string

	for _, val := range s.chain.GetValidators() {
		endpoint, err := val.GetTMClient(context.Background())
		s.Require().NoError(err)

		url := strings.Replace(endpoint.Remote(), "http", "ws", -1)

		endpoints = append(endpoints, fmt.Sprintf("%s/websocket", url))
	}

	ec := AMakeEncodingConfig()

	cf, err := petriload.NewDefaultClientFactory(
		petriload.ClientFactoryConfig{
			Chain:                 s.chain,
			Seeder:                s.user1,
			WalletConfig:          s.spec.WalletConfig,
			AmountToSend:          1000000,
			SkipSequenceIncrement: false,
			EncodingConfig: cosmosutil.EncodingConfig{
				InterfaceRegistry: ec.InterfaceRegistry,
				Codec:             ec.Codec,
				TxConfig:          ec.TxConfig,
			},
			MsgGenerator: func(sender []byte) ([]sdk.Msg, petritypes.GasSettings, error) {
				return []sdk.Msg{
						banktypes.NewMsgSend(
							sdk.AccAddress(sender),
							sdk.AccAddress([]byte{1}),
							sdk.NewCoins(sdk.NewCoin(s.chain.GetConfig().Denom, sdkmath.NewInt(1))),
						),
					}, petritypes.GasSettings{
						Gas:         200000,
						PricePerGas: int64(0),
						GasDenom:    s.chain.GetConfig().Denom,
					}, nil
			},
		}, app.ModuleBasics)

	s.Require().NoError(err)

	err = loadtest.RegisterClientFactory("cosmos_bank_send", cf)
	s.Require().NoError(err)

	cfg := loadtest.Config{
		ClientFactory:        "cosmos_bank_send",
		Connections:          1,
		Endpoints:            endpoints,
		Time:                 60,
		SendPeriod:           1,
		Rate:                 350,
		Size:                 250,
		Count:                -1,
		BroadcastTxMethod:    "async",
		EndpointSelectMethod: "supplied",
		StatsOutputFile:      "./test.csv",
	}
	err = loadtest.ExecuteStandalone(cfg)
	s.Require().NoError(err)
```

# Questions

- what's the best way to surface the gas required for a bank send (the initial seeding transaction) up to the load test? should it be hardcoded on the assumption that banksends will take the same amount of gas on all chains?

### CODE REVIEW

File: loadtest/client_factory.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+type ClientFactoryConfig struct {
+	Chain                 petritypes.ChainI
+	Seeder                *cosmosutil.InteractingWallet
+	WalletConfig          petritypes.WalletConfig
+	AmountToSend          int64
+	SkipSequenceIncrement bool
+	EncodingConfig        cosmosutil.EncodingConfig
+	MsgGenerator          GenerateMsgs
```
Comment (nivasan1) [inline loadtest/client_factory.go:35]: Is this caller meant to implement this to determine what msgs to send?

File: loadtest/client.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	petritypes "github.com/skip-mev/petri/types/v2"
+	petriutil "github.com/skip-mev/petri/util/v2"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type DefaultClient struct {
+	loader           *cosmosutil.InteractingWallet
+	chainClient      *cosmosutil.ChainClient
+	seq, accNum      uint64
```
Comment (nivasan1) [inline loadtest/client.go:20]: where do we set these?

**Comment by Zygimantass** [comment]
need to rebase on main before merging, rebased on v1.x.x to test with BSDK

File: chain/go.mod [hunk 3]
```diff
@@ -2,166 +2,130 @@ module github.com/skip-mev/petri/chain
...
-	github.com/skip-mev/petri/provider => ../provider
-	github.com/skip-mev/petri/util => ../util
-	github.com/skip-mev/petri/wallet => ../wallet
-)
-
 require (
 	cosmossdk.io/math v1.2.0
-	github.com/cometbft/cometbft v0.38.2
-	github.com/cosmos/cosmos-sdk v0.50.2
+	github.com/cometbft/cometbft v0.37.4
```
Comment (aljo242) [inline chain/go.mod:7]:
```suggestion
	github.com/cometbft/cometbft v0.37.5
```

File: loadtest/client.go (hunk 2 above)
Comment (Zygimantass) [reply loadtest/client.go:20]: fixed

File: loadtest/client.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+	gasSettings      petritypes.GasSettings
+	msgs             []sdk.Msg
```
[the remark this answers is not in the document]
Comment (Zygimantass) [reply loadtest/client.go:22]: will resolve in a later update

**Pull request merged**